packages/web/src/features/translation/TranslationView.tsx

```diff
@@ -209,10 +216,15 @@ export default function TranslationView() {
 
       alert('Unknown error occurred.');
     },
-    onSettled: (_, __, ___, context) => {
+    onSettled: (_, __, variables, context) => {
       queryClient.invalidateQueries({
         queryKey: ['verse-glosses', language, verseId],
       });
+      if (variables.state) {
+        queryClient.invalidateQueries({
+          queryKey: ['language-progress', language],
+        });
+      }
```

Comment:
We'll want to add this to the approve all mutation as well

Reply:
Done.